src/bt-pan-io.c

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright (C) 2015  Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <assert.h>
+#include <fdio/task.h>
+#include <hardware/bluetooth.h>
+#include <hardware/bt_pan.h>
+#include <stdlib.h>
+#include "compiler.h"
+#include "log.h"
+#include "bt-proto.h"
+#include "bt-pdubuf.h"
+#include "bt-core-io.h"
+#include "bt-pan-io.h"
+
+#define IFNAME_LEN 17 /* interface name */
```

Comment:
Hi Thomas,

I agree the static const is better than define for constant declaration. However, the C compiler cannot treat a const as a compile-time constant.  Therefore, if we changed to |static const|, we cannot declaration padding array like |static const uint8_t padding[IFNAME_LEN]|. If we do this, we will face the the error "size of array is not constant" when create array padding.